docs(weekly): add this week in databend 117 (#13516)
datafuselabs/databend: website/blog/2023-10-29-databend-weekly-117.md

---
title: "This Week in Databend #117"
date: 2023-10-29
slug: 2023-10-29-databend-weekly
cover_url: 'weekly/weekly-117.jpg'
image: 'weekly/weekly-117.jpg'
tags: [weekly]
description: "Stay up to date with the latest weekly developments on Databend!"
contributors:
  - name: akoshchiy
  - name: andylokandy
  - name: ariesdevil
  - name: b41sh
  - name: BohuTang
  - name: dantengsky
  - name: Dousir9
  - name: drmingdrmer
  - name: leiysky
  - name: PsiACE
  - name: SkyFan2002
  - name: soyeric128
  - name: sundy-li
  - name: Xuanwo
  - name: xudong963
  - name: youngsofun
  - name: zenus
  - name: zhang2014
  - name: zhyass
authors:
  - name: PsiACE
    url: https://github.com/psiace
    image_url: https://github.com/psiace.png
---

[Databend](https://github.com/datafuselabs/databend) is a modern cloud data warehouse, serving your massive-scale analytics needs at low cost and complexity. Open source alternative to Snowflake. Also available in the cloud: <https://app.databend.com> .

## What's New

Stay informed about the latest features of Databend.

> ### Feature Preview: Read-Only ATTACH TABLE

Reviewer: The feature is out now, as well as the document updates:) 
So I think we can get rid of "Feature Preview" from the title.

> ### Feature Preview: Read-Only ATTACH TABLE

Reviewer: ```suggestion
Read-Only ATTACH TABLE
```

> Maintaining large resources for just a few large-scale queries can bring high maintenance costs for enterprises. How can we mix local and cloud workloads, leverage Databend Cloud's instant scaling while ensuring query reliability, and further reduce the count of private instances to save on maintenance costs?

Reviewer: Maintaining extensive resources for just a few large-scale queries can lead to high maintenance costs for enterprises. How about we combine local and cloud workloads, harness Databend Cloud's instant scaling computing capabilities, all while avoiding the additional storage costs associated with cloud usage?

> Databend recently introduced **Read-only `ATTACH TABLE`**, allowing users with locally deployed Databend to share table data with Databend Cloud in read-only mode:

Reviewer: ```suggestion
Databend recently introduced the READ_ONLY argument to `ATTACH TABLE`**, allowing users with locally deployed Databend to share table data with Databend Cloud in read-only mode:
```

- All table data is shared in read-only mode.

> - Changes made to the table, including data and schema changes, will be performed and tracked on the user's local platform.

Reviewer: ```suggestion
- Changes made to a table, including data and schema changes, will be performed and tracked on the user's local platform.
```

Reviewer: I guess it is also true for non-read-only mode.

```sql
ATTACH TABLE <target_table_name> '<source-table-data-URI>' CONNECTION=(<connection_parameters>) READ_ONLY
```

> Adding the `READ_ONLY` parameter to the `ATTACH TABLE` statement will map the table data and schema using the latest snapshot hint file stored at the specified source table data URI, and the target table can only be accessed in a read-only manner.

Reviewer: 即使不是read-only, 也是用最新的snapshot

We refactored the original trace export mechanism using `opentelemetry-otlp` while fixing the availability of distributed tracing capabilities. The corresponding environment variable also changes to `DATABEND_OTEL_EXPORTER_OTLP_ENDPOINT`.

![img](../static/img/blog/databend-with-jaeger.jpg)

If you are interested in learning more, please check out the resources below:

- [Docs | Monitor - Jaeger](https://databend.rs/doc/monitor/tools/jaeger)
- [PR #13381 | feat: migrate tracing reporter from jaeger agent to opentelemetry collector (otlp)](https://github.com/datafuselabs/databend/pull/13381/)

## Highlights

We have also made these improvements to Databend that we hope you will find helpful:

- Added new aggregate function `quantile_tdigest_weighted`.
- Recluster supports distributed execution.

> - `DELETE` supports table aliases.

Reviewer: Can u share the pull request for this feature?

- SQLsmith supports `REPLACE INTO` and `HINT`. Read the latest blog [Leveraging Randomized Testing with SQLsmith to Uncover Bugs](https://databend.rs/blog/2023-10-24-sqlsmith-hero) to learn about Databend's SQLsmith design.
- Added JSON operators `->`, `->>`, `#>` and `#>>`. See [Docs | JSON Operators](https://databend.rs/doc/sql-functions/json-operators/) for details.

## What's Up Next

We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.

### Support for Defining CONNECTION

To more elegantly handle the mapping between role access control models in compute cluster instances and storage services, Databend plans to provide the ability to define and manage `CONNECTION`s.

> **CONNECTION** can hold identity credentials and access management (IAM) entities generated by external storage services, providing convenient and consistent access control experiences. Introducing **CONNECTION** can also simplify complex configurations in creating `STAGE` etc.

Reviewer: ```suggestion
**CONNECTION** can hold identity credentials and access management (IAM) entities generated by external storage services, providing convenient and consistent access control experiences. Introducing **CONNECTION** can also simplify complex statements that includes a connection to object storages, such as creating an external stage, etc.
```

> The referenced syntax specification is as follows:

Reviewer: ```suggestion

```
